plugins/catalog-react/src/components/GroupListPicker/GroupListPicker.tsx

```diff
@@ -0,0 +1,152 @@
+
+const useStyles = makeStyles({
+  btn: {
+    backgroundColor: 'transparent',
+    border: 'none',
+    margin: 0,
+    padding: 0,
+    width: '100%',
+  },
+  title: {
+    fontSize: '24px',
+    fontStyle: 'normal',
+    fontWeight: 700,
+    letterSpacing: '-0.25px',
+    lineHeight: '32px',
+    marginBottom: 0,
+  },
+});
+
+/**
+ * Props for {@link GroupListPicker}.
+ *
+ * @public
+ */
+export type GroupListPickerProps = {
+  placeholder: string;
+  groupTypes: Array<string>;
+  defaultGroup?: string;
+};
+
+/** @public */
+export const GroupListPicker = (props: GroupListPickerProps) => {
+  const classes = useStyles();
+  const catalogApi = useApi(catalogApiRef);
+
+  const { placeholder, groupTypes, defaultGroup = '' } = props;
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const [inputValue, setInputValue] = React.useState('');
+  const [group, setGroup] = React.useState(defaultGroup);
+
+  const handleClick = (event: React.MouseEvent<any, MouseEvent>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popover' : undefined;
+
+  const {
+    loading,
+    error,
+    value: groups,
+  } = useAsync(async () => {
+    const groupsList = await catalogApi.getEntities({
+      filter: {
+        kind: 'Group',
+        'spec.type': groupTypes,
+      },
+    });
+
+    return groupsList.items as GroupEntity[];
+  }, [catalogApi]);
+
+  if (error) {
+    return <ResponseErrorPanel error={error} />;
+  }
+
+  return (
+    <>
+      <Popover
+        anchorEl={anchorEl}
+        open={open}
+        onClose={handleClose}
+        anchorOrigin={{ horizontal: 'left', vertical: 'bottom' }}
+      >
+        <Autocomplete
+          data-testid="group-list-picker-input"
+          area-aria-describedby={id}
+          loading={loading}
+          options={groups ?? []}
+          groupBy={option => option.spec.type}
+          getOptionLabel={option => option.spec.profile?.displayName ?? ''}
+          inputValue={inputValue}
+          onInputChange={(_, value) => setInputValue(value)}
+          onChange={(_, newValue) => {
+            if (newValue) {
+              setGroup(newValue.spec.profile?.displayName ?? '');
+            }
+            setInputValue('');
+          }}
+          style={{ width: '200px' }}
+          renderInput={params => (
+            <TextField
+              {...params}
+              placeholder={placeholder}
+              variant="outlined"
+            />
+          )}
+        />
+      </Popover>
+      <button
+        id={id}
+        style={{ cursor: 'pointer' }}
+        onClick={handleClick}
+        className={classes.btn}
+        data-testid="group-list-picker-button"
+      >
+        <Box display="flex" flexDirection="row" alignItems="center">
+          <PeopleIcon fontSize="large" style={{ marginRight: '8px' }} />
```

Comment:
just pointing out because I'm already at it - this could be smth. like

```suggestion
          <PeopleIcon fontSize="large" style={(theme: BackstageTheme) => ({ marginRight: theme.spacing(1) }) } />
```

not to important - [here is the doc](https://v4.mui.com/customization/spacing/)